app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php

```diff
@@ -51,62 +52,50 @@ public function getAllCountries()
         return $activeCountries;
     }
 
-    public function projectsCountry($country)
+    public function numberOfProjects($country, $projects)
     {
-        $query = Project::where('framework_key', 'terrafund')
-            ->whereHas('organisation', function ($query) {
-            $query->whereIn('type', ['for-profit-organization', 'non-profit-organization']);
-        });
-        return $query->where('country', $country);
+        return $projects->where('country', $country)->count();
     }
 
-    public function numberOfProjects($country)
+    public function totalSpeciesAmount($country, $projects)
     {
-        return $this->projectsCountry($country)->count();
-    }
+        $projects = $projects->where('country', $country);
 
-    public function totalSpeciesAmount($country)
-    {
-        $projects = $this->projectsCountry($country)->get();
-        $totalSpeciesAmount = 0;
-        foreach ($projects as $project) {
-            $sites = $project->sites()->pluck('id')->toArray();
-            foreach ($sites as $siteId) {
-                $latestSiteReportId = SiteReport::where('site_id', $siteId)
-                    ->orderByDesc('due_at')
-                    ->value('id');
-                if ($latestSiteReportId !== null) {
-                    $totalSpeciesAmount += TreeSpecies::where('speciesable_id', $latestSiteReportId)->sum('amount');
+        return $projects->sum(function ($project) {
+            return $project->sites()->get()->sum(function ($site) {
```

Comment:
The `sum` function on `$projects` I think isn't working the way you want. That's a call on the query builder, which can't take a closure, only a [column name string](https://laravel.com/api/7.x/Illuminate/Database/Query/Builder.html#method_sum). Instead, you want `sum` on the collection, like you have on this line. So, I think for this to function correctly, it needs a `get()` in both cases.

```
return $projects->get()->sum(function($project) {
  return $project->sites()->get()->sum(function $site) {
```

However, that's running into N+ query land again, so a better approach would be to avoid the double loop:
```
return Sites::whereIn('project_id', $projects->pluck('id'))->get()->sum(function ($site) {
```